src/lib/rest/rest.cpp

```diff
@@ -1781,6 +1782,7 @@ void restInit
   mhdConnectionTimeout = _mhdTimeoutInSeconds;
 
   strncpy(restAllowedOrigin, _allowedOrigin, sizeof(restAllowedOrigin));
+  corsEnabled = (restAllowedOrigin[0] != 0);
```

Comment:
A cryptic way of checking if the char string is empty but it's ok... :)

Just to be sure... how `restAllowedOrigin` array is initialized? Can we asure that first element is zero? Or it is random and could be initilizalied with random stuff?

Reply:
Actually it was suggested by @kzangeli :) I think it is better than checking the length.

`restAllowedOrigin` gets the value of -corsOrigin switch or if the switch is not used, initialized with empty string, `""`. For empty string, the first char will always be zero.

By the way, since we are talking about the variable, we have re-used some of the code Jose Manuel wrote for the first CORS implementation and some of the variable names that made sense back then, now got a bit confusing. Especially restAllowedOrigin. So this is the flow we have now:

1. We take the `-corsOrigin` switch value and assign it to `allowedOrigin` in `contextBroker.cpp`
2. We call the restInit function, passing the value of `allowedOrigin` variable in `contextBroker.cpp`
3. We define a global variable in `rest.cpp` called `restAllowedOrigin`
4. In `restInit` function, the passed value is mapped to `_allowedOrigin` in the signature and the value is re-assigned to `restAllowedOrigin`. 

As you see, there are a lot of name changes which makes the code hard to read and harder to maintain. I propose using the same name corsOrigin except the method signature of `restInit` where we can have it as `_corsOrigin`.  Same goes for restCORSMaxAge.

Reply:
> if the switch is not used, initialized with empty string, ""

That's the piece of the puzzle I missed. I didn't know if the argument parsing library puts `""` in the case of missing swich or just left the holding variable with undefined content. I understand is the former.

> I propose using the same name corsOrigin except the method signature of restInit where we can have it as _corsOrigin. Same goes for restCORSMaxAge.

Sounds good.

Reply:
I guess this is related with

```
{ "-corsOrigin",       allowedOrigin,     "ALLOWED_ORIGIN",    PaString, PaOpt, _i "",          PaNL,  PaNL,     ALLOWED_ORIGIN_DESC    },
```
That `_i ""` ensures that in the case of not using `-corsOrigin` at the CLI, the value for allowedOrigin is univocally set to empty string, isn't it @kzangeli  ?

Reply:
That's it

Reply:
Ok. So this comment (and with it the whole PR, as far as I remember) is only pending of a new commit with the refactor in the variable names, as @McMutton suggested above.

Reply:
Fixed in e3a6471